### PR TITLE
features and unit tests for suppliers, products and consignments

### PR DESCRIPTION
**Pull request by pulkitsinghal**
please code review commit by commit for best results

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -450,25 +450,25 @@ describe('vend-nodejs-sdk', function () {
...
-          'sku': faker.fake('{{random.number}}'), // faker.random.number,
-          'supply_price': faker.fake('{{commerce.price}}') // faker.commerce.price
+          'name': faker.commerce.productName(), // REQUIRED
+          'sku': sku, // REQUIRED
+          'handle': faker.lorem.word(1), // REQUIRED
+          'retail_price': cost + 5.00, // REQUIRED
+          'supply_price': cost,
+          'description': faker.lorem.sentence()
         };
         randomProduct.price = String(Number(randomProduct['supply_price']) + 10.00);
```
Comment (kamal0808) [inline test/test.js:612]: I couldn't find a `price` attribute here https://docs.vendhq.com/v0.9/reference#createupdateproduct-1. Was this documented somewhere else?

File: lib/consignment.js [hunk 2]
```diff
@@ -506,14 +505,12 @@ var createStockOrder = function(args, connectionInfo, retryCounter) {
...
   var authString = 'Bearer ' + connectionInfo.accessToken;
-  log.silly('Authorization: ' + authString); // TODO: sensitive data ... do not log?
 
   var body = {
     'type': 'SUPPLIER',
     'status': 'OPEN',
     'name': args.name.value,
-    'date': moment().format('YYYY-MM-DD HH:mm:ss'), //'2010-01-01 14:01:01',
-    'due_at': args.dueAt.value,
+    'due_at': args.dueAt.value, // caller must format it as `YYYY-MM-DD HH:mm:ss', example: '2010-01-01 14:01:01'
```
Comment (kamal0808) [inline lib/consignment.js:520]: I couldn't find this attribute in the tests. But the test still doesn't fail for some reason.

**Comment by kamal0808** [comment]
I did a code review based on code quality and placement of blocks of code. Everything looks good to me, can be merged immediately.

**Comment by kamal0808** [comment]
The only thing I'm not sure about is why we are doing all these in a `v2` feature branch.

File: test/test.js (hunk 1 above)
Comment (pulkitsinghal) [reply test/test.js:612]: seem like a mistake ... i went through the docs to add `retail_price` in this PR but didn't notice that line ... good catch ... @tinker20 do you also agree?

**Comment by pulkitsinghal** [comment]
> The only thing I'm not sure about is why we are doing all these in a v2 feature branch.

The v2 feature branch was created to add v2 apis but after a while, I couldn't do important work without cleanup so it suddenly changed into refactoring work to ensure previous tests are sufficient and code properly partitioned ... so the v2 code can be tested as a drop-in replacement of the previous APIs in tests ... as and when it is being added

would love to somehow cleverly merge the non 2.x parts back to `develop` too

File: lib/consignment.js (hunk 2 above)
Comment (pulkitsinghal) [reply lib/consignment.js:520]: there is no reason for the test to fail if the user doesn't specify this

File: test/test.js (hunk 1 above)
Comment (pulkitsinghal) [reply test/test.js:612]: i'll merge this PR and let's file a separate issue for this

**Pull request merged**